src/playlist-maker.js

```diff
@@ -2,6 +2,60 @@ import videojs from 'video.js';
 import playItem from './play-item';
 import * as autoadvance from './auto-advance';
 
+/**
+ * Generate a unique id for each playlist item object. This id will be used to determine
+ * index of an item in the playlist array for cases where there are multiple items with
+ * the same source set.
+ *
+ * @private
+ *
+ * @param {Array}
+ *         An array of playlist items
+ *
+ * @return {Array}
+ *          The list of playlist items with unique ids
+ */
+const generatePlaylistItemId = (arr) => {
+  const list = [];
+
+  for (let i = 0; i < arr.length; i++) {
+    const uuid = Date.now().toString(36) + Math.random().toString(36).substr(2, 5);
+
+    if (typeof arr[i] === 'object') {
```

Comment:
maybe it's time to normalize the input to the playlist plugin so that it's always an object?
Also, this may break if someone passes an array with `null` to playlist as they're objects but not one you can attach properties to.